resteasy/src/main/java/com/cerner/beadledom/resteasy/ResteasyContextListener.java

```diff
@@ -76,4 +70,21 @@ protected void withInjector(Injector injector) {
    * Returns the list of Guice modules to be used with Resteasy.
    */
   protected abstract List<? extends Module> getModules(ServletContext context);
+
+  /**
+   * Processes module bindings in the Guice injector.
+   */
+  @VisibleForTesting
+  public void processInjector(ServletContext context, Injector injector) {
```

Comment:
The other way to detect a loop is to use  run a fast-slow runner technique to detect the loop in a LinkedList. Seems like a overkill but may be worth trying it.